packages/core/src/types/config.ts

```diff
@@ -1,6 +1,7 @@
-export type TaskConfig = 'on' | 'off' | ['on' | 'off', unknown];
+export type ActionConfig = { [key: string]: number | 'off' };
```

Comment:
I think this should be changed to:

```suggestion
export type ActionConfig = { [key: string]: 'on' | 'off' | { threshold: number } };
```

I think this accounts for the following cases:
- For actions that are off by default (I know we discussed on by default, but we can't guarantee that all authors of tasks won't want to turn off tasks), you can turn them 'on'
- For actions that are on by default, you can turn them 'off'
- For actions that you want to configure the threshold for, you should explicitly set it, so that future readers of the config will know what that number represents without having to look it up in docs/read the code.

Reply:
We actually can guarantee that authors of tasks wont turn off actions by default, because we removed the property where that could be set and that feature is not supported. The `ActionList` does not look for the word `on` and there isnt a property that can be set on an Action to turn it off. So there is no reason to account for `on` here, unless you are changing your mind and and want to allow tasks to be turned off by default (which I had originally in this PR as `enabledByDefault`). I can put that back in if that is what you are suggesting, but confused as to what else you could be referring to since we aligned that for this initial iteration, all tasks would be turned on by default

Reply:
Made the change to have threshold be an object instead of a raw number - agreed this is better for readability